.github/workflows/ci.yml

```diff
@@ -58,12 +58,29 @@ jobs:
   build:
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-latest, windows-latest]
+        target: [x86_64-unknown-linux-gnu, armv7-unknown-linux-gnueabihf, x86_64-apple-darwin, x86_64-pc-windows-gnu]
         include:
-          - os: ubuntu-latest
-            e2e: true
-          - os: macos-latest
-            e2e: true
+
+          - target: x86_64-unknown-linux-gnu
+            os: ubuntu-latest
+            test_features: testcontainers
+
+          - target: armv7-unknown-linux-gnueabihf
+            os: ubuntu-latest
+            # This target is only for cross-compiling, skip all tests.
+            skip_unit_tests: true
+            skip_e2e_test: true
+
+          - target: x86_64-apple-darwin
+            os: macos-latest
+            e2e_tests: true
```

Comment:
I guess the idea is to remove this attribute and only have the `skip_e2e_test` one?

Reply:
Yes, I thought it is more readable if the matrix highlights the exceptions we are making per platform, rather than what we want to do and from there you have to infer what happens when.

Reply:
My comment was more about the odd attribute than the approach :)

Reply:
Yes thanks for pointing that out!